src/ui/public/courier/data_source/_highlight_body.js

```diff
@@ -0,0 +1,45 @@
+import _ from 'lodash';
+import 'ui/highlight/highlight_tags';
+
+const FRAGMENT_SIZE = Math.pow(2, 31) - 1; // Max allowed value for fragment_size (limit of a java int)
+
+/**
+ * Generates a `highlight_query` corresponding to the given query by setting `"all_fields": true` on any found `query_string` queries
+ */
+function getHighlightQuery(query) {
+  const clone = _.cloneDeep(query);
+
+  if (_.has(clone, 'query_string')) {
+    clone.query_string.all_fields = true;
+  } else if (_.has(clone, 'bool.must')) {
+    clone.bool.must = clone.bool.must.map(getHighlightQuery);
```

Comment:
I don't see a way to reliably do this without transforming the query into a normal form first either (which would be equivalent to an "intermediate representation"). :frowning_face: